Jenkinsfile.deploy

```diff
@@ -42,37 +56,118 @@ pipeline {
                     if (params.DISPLAY_NAME != '') {
                         currentBuild.displayName = params.DISPLAY_NAME
                     }
+
+                    if (isRelease()) {
+                        assert getProjectVersion() != ''
+                    }
                 }
             }
             post {
                 success {
                     script {
                         setDeployPropertyIfNeeded('git.branch', getBuildBranch())
                         setDeployPropertyIfNeeded('git.author', getGitAuthor())
+                        setDeployPropertyIfNeeded('project.version', getProjectVersion())
+                        setDeployPropertyIfNeeded('release', isRelease())
                     }
                 }
             }
         }
+
         stage('Clone repositories') {
             steps {
                 checkoutRepo('optaplanner')
             }
         }
+
+        stage('Prepare for PR'){
+            when {
+                expression { return isRelease() }
+            }
+            steps {
+                prepareForPR('optaplanner')
+            }
+        }
+
+        stage('Setup Maven release config'){
+            when {
+                expression { return isRelease() }
+            }
+            steps {
+                echo 'Setup Maven release config'
+                configFileProvider([configFile(fileId: maven.getSubmarineSettingsXmlId(), targetLocation: 'maven-settings.xml', variable: 'MAVEN_SETTINGS_FILE')]){
+                    sh "echo '-B -s ${MAVEN_SETTINGS_FILE}' | tee optaplanner/.mvn/maven.config"
+                    // TODO: figure out why the assert does not work
+                    // assert params.MAVEN_DEPENDENCIES_REPOSITORY != '' // OptaPlanner depends on Kogito Runtimes
```

Comment:
this parameter should not be mandatory. runtimes could have been released already. thus this pipeline would not need any `MAVEN_DEPENDENCIES_REPOSITORY`

Reply:
just thinking I need to update that as well on examples that we should check `params.MAVEN_DEPENDENCIES_REPOSITORY` is not empty before doing the change ...

Reply:
already implemented in fact on examples.See https://github.com/kiegroup/kogito-examples/blob/master/Jenkinsfile.deploy#L97

Reply:
Good point, will remove the assert and add a condition.